pinot-segment-local/src/main/java/org/apache/pinot/segment/local/realtime/impl/json/MutableJsonIndexImpl.java

```diff
@@ -291,6 +297,88 @@ private RoaringBitmap getMatchingFlattenedDocIds(Predicate predicate) {
       } else {
         return new RoaringBitmap();
       }
+    } else if (predicateType == Predicate.Type.REGEXP_LIKE) {
+      String ceilingKey = _postingListMap.ceilingKey(key + BaseJsonIndexCreator.KEY_VALUE_SEPARATOR);
+      if (ceilingKey == null || !ceilingKey.startsWith(key + BaseJsonIndexCreator.KEY_VALUE_SEPARATOR)) {
+        // No values with this key exist
+        return new RoaringBitmap();
+      }
+
+      Pattern pattern = ((RegexpLikePredicate) predicate).getPattern();
+      MutableRoaringBitmap result = null;
+      char nextChar = BaseJsonIndexCreator.KEY_VALUE_SEPARATOR + 1;
+      SortedMap<String, RoaringBitmap> subMap =
+          _postingListMap.subMap(ceilingKey, true, _postingListMap.floorKey(key + nextChar), true);
+
+      for (Map.Entry<String, RoaringBitmap> entry : subMap.entrySet()) {
+        if (!pattern.matcher(entry.getKey().substring(key.length() + 1)).matches()) {
+          continue;
+        }
+        if (result == null) {
+          result = entry.getValue().toMutableRoaringBitmap();
```

Comment:
Suggest not converting it to `MutableRoaringBitmap`. `RoaringBitmap` itself is mutable, and is faster

Reply:
Ack